modules/base/dashboard/dashboarditemtimevaryingtext.cpp

```diff
@@ -0,0 +1,161 @@
+#include <openspace/util/timemanager.h>
+#include <openspace/json.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/logging/logmanager.h>
+
+
+
+namespace {
+    constexpr std::string_view _loggerCat = "DashboardItemTimeVaryingText";
+
+    constexpr openspace::properties::Property::PropertyInfo TextBeforeInfo = {
+        "TextBefore",
+        "Text Before",
+        "Optional text before the time varying text to be displayed.",
+        openspace::properties::Property::Visibility::User
+    };
+    constexpr openspace::properties::Property::PropertyInfo DataFileInfo = {
+        "DataFile",
+        "Data File Path",
+        "The file path to the JSON data.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    struct [[codegen::Dictionary(DashboardItemTimeVaryingText)]] Parameters {
+        // [[codegen::verbatim(TextBeforeInfo.description)]]
+        std::optional<std::string> textBefore;
+        // [[codegen::verbatim(DataFileInfo.description)]]
+        std::string dataFile;
+    };
+#include "dashboarditemtimevaryingtext_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation DashboardItemTimeVaryingText::Documentation() {
+    return codegen::doc<Parameters>(
+        "base_dashboarditem_timevaryingtextBefore",
+        DashboardTextItem::Documentation()
+    );
+}
+
+DashboardItemTimeVaryingText::DashboardItemTimeVaryingText(
+                                                      const ghoul::Dictionary& dictionary)
+    : DashboardTextItem(dictionary)
+    , _textBefore(TextBeforeInfo, "")
+    , _dataFile(DataFileInfo, "")
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+    _textBefore = p.textBefore.value_or(_textBefore);
+    addProperty(_textBefore);
+    _dataFile = absPath(p.dataFile).string();
+    addProperty(_dataFile);
+    loadDataFromJson(_dataFile);
+}
+
+void DashboardItemTimeVaryingText::update() {
+    if (_startTimes.empty()) {
+        _buffer = "";
+        return;
+    }
+
+    double current = global::timeManager->time().j2000Seconds();
+    double first = _startTimes.front();
+    double last = _sequenceEndTime;
+
+    if (current >= first && current < last) {
+        int newIdx = updateActiveTriggerTimeIndex(current);
+        if (newIdx != _activeTriggerTimeIndex) {
+            _activeTriggerTimeIndex = newIdx;
+            double timeKey = _startTimes[_activeTriggerTimeIndex];
+            double value = _data[timeKey];
+
+            std::ostringstream oss;
+            oss << value;
+            _buffer = _textBefore.value() + oss.str();
+        }
+    }
+    else {
+        _activeTriggerTimeIndex = -1;
+        _buffer = "";
+    }
+}
+
+void DashboardItemTimeVaryingText::loadDataFromJson(const std::string& filePath) {
+    std::ifstream file(filePath);
+    if (!file.is_open()) {
+        LERROR(std::format(
+            "Time varying text, '{}' is not a valid JSON file",
+            filePath
+        ));
```

Comment:
```suggestion
        throw ghoul::Runtime(std::format(
            "Time varying text, '{}' is not a valid JSON file",
            filePath
        ));
```

Since it is not a recoverable error if the JSON file doesn't load